truffle/src/main/ruby/core/rubinius/api/shims/mirror.rb

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2015 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+module Rubinius
+  class Mirror
+
+    # Implementation of the :module_mirror Rubinius primitive.
```

Comment:
I'm not sure what you're asking.  But `:module_mirror` is used by Rubinius to get the mirror for an object.  We had an incomplete implementation in Java.  This is a complete implementation written in Ruby.  The Java one now just calls this code.